packages/jellyfish-api-core/__tests__/category/loan/createVault.test.ts

```diff
@@ -0,0 +1,227 @@
+    expect(data).toStrictEqual({
+      loanSchemeId: 'scheme',
+      ownerAddress: GenesisKeys[0].owner.address,
+      isUnderLiquidation: false,
+      collateralAmounts: [],
+      loanAmount: [],
+      collateralValue: expect.any(String),
+      loanValue: expect.any(String)
+    })
+  })
+
+  it('should not createVault with utxos not from the owner', async () => {
+    const utxo = await testing.container.fundAddress(await testing.generateAddress(), 10)
+    const promise = testing.rpc.loan.createVault({
+      ownerAddress: GenesisKeys[0].owner.address,
+      loanSchemeId: 'scheme'
+    }, [utxo])
+
+    const sBuff = new SmartBuffer()
+    OP_CODES.toBuffer(P2PKH.fromAddress(RegTest, GenesisKeys[0].owner.address, P2PKH).getScript().stack, sBuff)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test VaultTx execution failed:\ntx must have at least one input from token owner ' + sBuff.toString('hex').substring(2) + '\', code: -32600, method: createvault')
+  })
+})
+
+describe('Loan createVault with scheme set to be destroyed', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should not createVault with scheme set to be destroyed', async () => {
+    // Default scheme
+    await testing.rpc.loan.createLoanScheme({
+      minColRatio: 100,
+      interestRate: new BigNumber(1.5),
+      id: 'default'
+    })
+    await testing.generate(1)
+
+    // create another scheme "scheme"
+    await testing.rpc.loan.createLoanScheme({
+      minColRatio: 200,
+      interestRate: new BigNumber(2.5),
+      id: 'scheme'
+    })
+    await testing.generate(1)
+
+    // Wait for block 110
+    await testing.container.waitForBlockHeight(110)
+
+    // To delete at block 120
+    await testing.rpc.loan.destroyLoanScheme({ id: 'scheme', activateAfterBlock: 120 })
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.createVault({
+      ownerAddress: await testing.generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test VaultTx execution failed:\nCannot set scheme as loan scheme, set to be destroyed on block 120\', code: -32600, method: createvault')
+  })
+})
+
+describe('Loan createVault when no default scheme and CreateVault.loanSchemeId is not given', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should not createVault when no default scheme and CreateVault.loanSchemeId is not given', async () => {
```

Comment:
move it on top of `it('should not createVault with scheme set to be destroyed'`

Reply:
if we move this test to the same container as `should not createVault with scheme set to be destroyed`,
if executed as the first `it()`, should be ok,  but executed second, it will fail. tests should be able to execute in any order, 
also once you set the default scheme, it can not be deleted.
Hence better to be in a separate container.